change-address/REPWRITERSPECS/BANNO.CHANGE.ADDR.V1.POW

```diff
@@ -276,6 +306,475 @@ PRINT TITLE="BANNO.CHANGE.ADDR.V1.POW"
  NEWLINE
 END [PRINT]
 
+PROCEDURE NAMEFMBYLEVEL
+[* Performs file maintenance on all selected name records and 
+** allowed name levels. Only called when old address matching is 
+** enabled.
+*]
+
+[ Account level name FM]
+ IF MATCHACCTNAMES=TRUE AND
+    ACCTNAMECOUNT>0 AND
+    RESPONSECODE=0 THEN
+  DO
+   FOR TMPLOOP=1 TO ACCTNAMECOUNT
+    DO
+     FMPERFORM REVISE NAME LOC ACCTLVLLOCS(TMPLOOP)(0,0,FMERROR)
+      DO
+       SET EXTRAADDRESS TO UPPERCASE(NEWEXTRAADDR)
+       SET STREET TO UPPERCASE(NEWSTREET)
+       SET CITY TO UPPERCASE(NEWCITY)
+       SET STATE TO UPPERCASE(NEWSTATE)
+       SET ZIPCODE TO UPPERCASE(NEWZIPCODE)
+      END
+
+     BNODEBUGMSG=FORMAT("ACCT NAME LOC #####9 TARGETED. ",ACCTLVLLOCS(TMPLOOP))+"FMERROR='"+FMERROR+"'"
+     CALL BNODEBUGMSGADD
+
+     IF FMERROR<>"" THEN
+      DO
+       RESPONSECODE=503
+       ERROR503CHR=FORMAT("Banno Chg Addr Err - Nm LOC #####9",ACCTLVLLOCS(TMPLOOP))+" FM-"+FMERROR
+       PARSEDTEXT(1)=FORMAT("Banno Chg Addr Err - Nm LOC #####9",ACCTLVLLOCS(TMPLOOP))
+       PARSEDTEXT(2)="FM-"+FMERROR
+       CALL CREATENOTERECORD
+      END
```

Comment:
It appears for each of these sections in the hypothetical situation that two or more names get an FM Error for a Name Type Update, that the Later "PROCEDURE RETURNJSON1" will only "CALL ERRORHANDLER" once showing what is in the variables for the Last Error .
Is this reasonable? Or should it stop looping after the first error?

Reply:
I have changed the FOR loops to WHILE loops and included the RESPONSECODE=0. This will stop the FM with any FM error, including during a loop.